File: MangaST_1_11/FrontMangaST/src/app/services/models/manga-response.ts
/* tslint:disable */
/* eslint-disable */
import { Category } from '../models/category';
import { CommentResponse } from '../models/comment-response';
export interface MangaResponse {
  achieved?: boolean;
  authorName?: string;
  categoryList?: Array<Category>;
  comments?: Array<CommentResponse>;
  description?: string;
  id?: number;
  mangaCover?: Array<string>;
  mangaPageId?: string;
  pathTitle?: string;
  rating?: number;
  status?: 'ANNOUNCE' | 'ONGOING' | 'PAUSED' | 'COMPLETED' | 'DISCONTINUED';
  title?: string;
  typeManga?: 'MANGA' | 'MANHWA';
}

File: MangaST_1_11/src/main/java/com/example/mangast/manga/MangaResponse.java
package com.example.mangast.manga;

import com.example.mangast.comments.Comment;
import com.example.mangast.comments.CommentResponse;
import com.example.mangast.manga.categories.Category;
import lombok.*;

import java.util.List;


@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class MangaResponse {

    private Integer id;

    private String title;

    private String authorName;

    private byte[] mangaCover;

    private boolean achieved;

    private String mangaPageId;

    private String pathTitle;

    private TypeManga typeManga;

    private MangaStatus status;

    private String description;

    private Double rating;

    private List<Category> categoryList;

    private List<CommentResponse> comments;


}
